app/services/metrics/merge_time/per_department.rb

```diff
@@ -0,0 +1,42 @@
+module Metrics
+  module ReviewTurnaround
+    class PerDepartment < Metrics::Base
+      def initialize(interval = nil)
+        @interval = interval
+      end
+
+      def call
+        process
+      end
+
+      private
+
+      def process
+        department_metrics_values.each do |department_name, metrics_values|
+          department_id = Department.find_by!(name: department_name)
+          turnaround = calculate_turnaround(metrics_values)
+          create_or_update_metric(department_id, Department.to_s, metric_interval,
+                                  turnaround, :review_turnaround)
+        end
+      end
+
+      def department_metrics_values
+        Department.names.keys.each_with_object({}) do |department_name, hash|
```

Comment:
Why not just iterate through the department objects, so that you don't have to find them in line 16

Reply:
Fixed!